proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+
+  * Return `true` if `$color` is a [legacy color], or `false`
+    otherwise.
+
+### `color.is-powerless()`
+
+* ```
+  is-powerless($color, $channel, $space: null)
+  ```
+
+  * If `$color` is not a color, throw an error.
+
+  * If `$space` is null:
+
+    * Let `color` be `$color`, and let `space` be the result of calling
+      `space($color)`.
+
+  * Otherwise:
+
+    * Let `color` be the result of calling `color.to-space($color, $space)`,
+      and let `space` be `$space`.
+
+  * If `space` is not a [known color space], throw an error.
+
+  * If `$channel` is not the name of a channel in the color-space `space`,
+    throw an error.
+
+  * Return `true` if the channel `$channel` is [powerless](#powerless-components)
+    in `color`, otherwise return `false`.
+
+### `color.is-in-gamut()`
+
+* ```
+  is-in-gamut($color, $space: null)
+  ```
+
+  * If `$color` is not a color, throw an error.
+
+  * Let `space` be the value of `$space` if specified, or the result of calling
+    `color.space($color)` otherwise.
+
+  * If `space` is not a [known color space], throw an error.
+
+  * Let `color` be the result of calling `color.to-space($color, space)`.
+
+  * For all bounded channels in `space`, if the associated channel value in
+    `$color` is outside the bounded range, return `false`.
+
+  * Otherwise, return `true`.
+
+### `color.to-gamut()`
+
+* ```
+  to-gamut($color, $space: null)
+  ```
+
+  * If `$color` is not a color, throw an error.
+
+  * Let `origin-space` be the result of calling `color.space($color)`.
+
+  * Let `target-space` be the value of `$space` if specified, or the value of
+    `origin-space` otherwise.
+
+  * If `target-space` is not a [known color space], throw an error.
+
+  * Return the result of [gamut mapping] with `$color` as the
+    origin color, `origin-space` as the origin color space, and
+    `target-space` as the destination color space.
+
+[gamut mapping]: #gamut-mapping
+
+### `color.channel()`
+
+> Note that channel values are stored as specified, even if those values are
+> out-of-gamut for the [known color space] used. Similarly, this color-channel
+> inspection function may return out-of-gamut channel values.
+
+* ```
+  channel($color, $channel, $space: null)
+  ```
+
+  * If `$space` is null:
+
+    * Let `space` be the result of calling `color.space($color)`, and let
+      `color` be the value of `$color`.
+
+  * Otherwise:
+
+    * Let `color` be the result of calling `color.to-space($color, $space)`,
+      and let `space` be the value of `$space`.
+
+  * Let `channels` be a map 1-indexed integer channel keys and their
+    corresponding values in `color`.
+
+  * If `space` is a [known color space]:
+
+    * Let `named-channels` be a map of channel names defined by `space`, and
+      their corresponding values in `color`.
+
+    * Set `channels` to the result of `map.merge(channels, named-channels)`.
+
+  * Let `value` be the result of calling `map.get(channels, $channel)`.
```

Comment:
We should probably also throw an error if `$channel` isn't a number or unquoted string.